pocketnetteam/pocketnet.core
# Sqlite DB cleanup, enable bench tests, and enable unittests on bitcoin upstream

Fix orphan transactions in SQLite database and add unitttest
- Clean expired transactions from SQLite Payload and Transactions db

- Fix error in SQLite query for orphaned transactions

- Enable mempool_tests

- Add verifydb function to validate sqlite transactions are present on the blockchain

- Additional debugging and check mempool for existance of transactions

- Re-arrange locks in LoadMemPool to be consistent with lock usage through rest of code

- Refactor CleanSQLite function and print error if transaction to delete is present in blocks or mempool

- Verify DB deletes orphan transactions and add removal reason to CleanSQLite logging

- Use GetTransaction call to verify TX is valid

- Fix segmentation fault caused by accessing mempool TX which has been removed

- Enable unittests txvalidation_tests, txvalidationcache_tests, and validation_block_tests

- Remove unused parameter from ProcessNewBlock

- Add test pocketnet_block_tests

- pocketnet_block_tests: test InvalidateBlock and adding block to chain

- Add transactions to block from mempool

- Fix execution of unittests and disable Segwit on regtest for now

- Fix pocketnet_block_tests error by setting up mocktime for tests

- Intermittent test failures were occuring because blocks were generated with same timestamp and same hash value due to one second time stamp granularity.  Now timestamp is incremented for each block.

- Enable wallet tests.
